addons/api/addon/utils/indexedDbQuery.js

```diff
@@ -0,0 +1,267 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import { get } from '@ember/object';
+import { modelIndexes } from '../services/indexed-db';
+
+/**
+ * Takes a POJO representing a filter query and builds a query to
+ * query indexedDb with. Note that the first filter is important as that
+ * will use the actual index in indexedDb while any subsequent ones will be less efficient.
+ *
+ * @example
+ *   const filter = buildIndexedDbQuery(indexedDb, type, {
+ *     search: {
```

Comment:
i think the function in this comment should be renamed to `queryIndexedDb`? Another thing, isn't the `query` param in this comment supposed to be something more like this `{ search: 'generated', filters: {...}` because the `search` object in this comment is created by the `getSearchObject` function.

Reply:
Yup you're right, I'll fix the example